pkg/deploy/controller/lifecycle/manager.go

```diff
@@ -0,0 +1,164 @@
+package lifecycle
+
+import (
+	"fmt"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+// Status represents the status of a lifecycle action, and should be treated
+// as the source of truth by controllers.
+type Status string
+
+const (
+	// Pending means the action has not yet executed.
+	Pending Status = "Pending"
```

Comment:
Okay, I need more coffee. Ignore everything I said before, and instead:

1. The annotations specific to the plugin (e.g. `PreExecNewPodActionPodPhaseAnnotation`) are in `deploy/api` and hold a `PodPhase`. These are also publicly exposed.
2. The status constants in `lifecycle` are *not stored* (currently), which is why they aren't publicly exposed. If they should be stored, I can expose them. I'm thinking they probably should be exposed, since there's not a 1:1 mapping between plugin status and the interpreted status which is used to actually drive the final behavior.